contracts/gateway/GatewayBase.sol

```diff
@@ -156,7 +159,7 @@ contract GatewayBase is Organized {
     /**
      *  @notice proveGateway can be called by anyone to verify merkle proof of
      *          gateway/co-gateway contract address. Trust factor is brought by
-     *          stateRoots mapping. stateRoot is committed in commitStateRoot
+     *          stateRoots mapping. stateRoot is anchored in anchorStateRoot
```

Comment:
`mapping. stateRoot is provided by stateRootProvider ...`
It should not be specific to anchor or core.
Most importantly, this comment doesn't make any sense anymore. The Anchor is by design **not** a decentralized mosaic process!